komodo-relational/src/main/java/org/komodo/relational/teiid/internal/CachedTeiidImpl.java

```diff
@@ -521,11 +557,76 @@ public Driver getDriver(UnitOfWork transaction, String name) throws KException {
         }
 
         return null;
     }
+
+    /* (non-Javadoc)
+     * @see org.komodo.relational.teiid.CachedTeiid#removeVdb(org.komodo.spi.repository.Repository.UnitOfWork, java.lang.String)
+     */
+    @Override
+    public void removeVdb(UnitOfWork transaction,
+                          String name) throws KException {
+        ArgCheck.isNotNull(transaction, "transaction"); //$NON-NLS-1$
+        ArgCheck.isTrue((transaction.getState() == State.NOT_STARTED), "transaction state is not NOT_STARTED"); //$NON-NLS-1$
+
+        KomodoObject folderNode = super.getChild(transaction, CachedTeiid.VDBS_FOLDER, KomodoLexicon.Folder.NODE_TYPE);
+        if(folderNode==null) {
+            return;
+        }
+
+        if (folderNode.hasChild(transaction, name, VdbLexicon.Vdb.VIRTUAL_DATABASE)) {
+            KomodoObject kobject = folderNode.getChild(transaction, name, VdbLexicon.Vdb.VIRTUAL_DATABASE);
+            kobject.remove(transaction);
+        }
+    }
+
+    /* (non-Javadoc)
+     * @see org.komodo.relational.teiid.CachedTeiid#removeDataSource(org.komodo.spi.repository.Repository.UnitOfWork, java.lang.String)
+     */
+    @Override
+    public void removeDataSource(UnitOfWork transaction,
+                                 String name) throws KException {
+        ArgCheck.isNotNull(transaction, "transaction"); //$NON-NLS-1$
+        ArgCheck.isTrue((transaction.getState() == State.NOT_STARTED), "transaction state is not NOT_STARTED"); //$NON-NLS-1$
+
+        KomodoObject folderNode = super.getChild(transaction, CachedTeiid.DATA_SOURCES_FOLDER, KomodoLexicon.Folder.NODE_TYPE);
+        if(folderNode==null) {
+            return;
+        }
+
+        if (folderNode.hasChild(transaction, name, DataVirtLexicon.Connection.NODE_TYPE)) {
+            KomodoObject kobject = folderNode.getChild(transaction, name, DataVirtLexicon.Connection.NODE_TYPE);
+            kobject.remove(transaction);
+        }
+    }
+
+    /* (non-Javadoc)
+     * @see org.komodo.relational.teiid.CachedTeiid#removeDriver(org.komodo.spi.repository.Repository.UnitOfWork, java.lang.String)
+     */
+    @Override
+    public void removeDriver(UnitOfWork transaction,
+                             String name) throws KException {
+        ArgCheck.isNotNull(transaction, "transaction"); //$NON-NLS-1$
+        ArgCheck.isTrue((transaction.getState() == State.NOT_STARTED), "transaction state is not NOT_STARTED"); //$NON-NLS-1$
+
+        KomodoObject folderNode = super.getChild(transaction, CachedTeiid.DRIVERS_FOLDER, KomodoLexicon.Folder.NODE_TYPE);
+        if(folderNode==null) {
+            return;
+        }
+
+        if (folderNode.hasChild(transaction, name, DataVirtLexicon.ResourceFile.DRIVER_FILE_NODE_TYPE)) {
+            KomodoObject kobject = folderNode.getChild(transaction, name, DataVirtLexicon.ResourceFile.DRIVER_FILE_NODE_TYPE);
+            kobject.remove(transaction);
+        }
+    }
```

Comment:
Sorry! Really don't like these methods. The CachedTeiid state is wholy dependent upon the state of the Teiid instance (otherwise its not a cache).